lib/tree.rb

```diff
@@ -1,25 +1,41 @@
 class NoApplesError < StandardError; end
 
-class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+class Tree
+  attr_reader :height, :age
+  MAXIMUM_AGE = 100
+  MAXIMUM_FRUIT_DIAMETER = 10
 
   def initialize
+    @height = 0.5
+    @age = 0
+    @apples = []
   end
 
   def age!
+    @age += 1
+    @height += 0.5 unless height >= 30
+    add_apples
   end
 
   def add_apples
+    if @age >= 3
+      rand(@age / 2...@age).times do
```

Comment:
I wanted a random number between `@age / 2` and `@age`. So if `@age = 50` this should resolve to `rand(25...50)`. Order of operations puts the division first.

This is just a nonsense algorithm for determining how many apples the tree should grow. I have no idea how real trees do it :P